fastjmd95/jmd95wrapper.py

```diff
@@ -6,42 +6,37 @@
 
 import fastjmd95.jmd95numba as jmd95numba
 
+def _any_dask_array(a,b,c):
+    any_dask = any([isinstance(a,dask.array.core.Array),
+           isinstance(b,dask.array.core.Array),
+           isinstance(c,dask.array.core.Array)])
+    return any_dask
 
-def rho_x(s, t, p):
-    if any([isinstance(s,dask.array.core.Array),
-           isinstance(t,dask.array.core.Array),
-           isinstance(p,dask.array.core.Array)]):
-        rho = dsa.map_blocks(jmd95numba.rho,s,t,p)
-    elif any([[isinstance(s,xr.DataArray),
-           isinstance(t,xr.DataArray),
-           isinstance(p,xr.DataArray)]]):
-        rho = xr.apply_ufunc(jmd95numba.rho,s,t,p,dask='allowed')
-    else:
-        rho = jmd95numba.rho(s,t,p)
-    return rho
+def _any_xarray(a,b,c):
+    any_xarray = any([[isinstance(a,xr.DataArray),
+           isinstance(b,xr.DataArray),
+           isinstance(c,xr.DataArray)]])
+    return any_xarray
 
-def drhodt_x(s, t, p):
-    if any([isinstance(s,dask.array.core.Array),
-           isinstance(t,dask.array.core.Array),
-           isinstance(p,dask.array.core.Array)]):
-        drhodt = dsa.map_blocks(jmd95numba.drhodt,s,t,p)
-    elif any([[isinstance(s,xr.DataArray),
-           isinstance(t,xr.DataArray),
-           isinstance(p,xr.DataArray)]]):
-        drhodt = xr.apply_ufunc(jmd95numba.drhodt,s,t,p,dask='allowed')
-    else:
-        drhodt = jmd95numba.rho(s,t,p)
-    return drhodt
+def my_decorator(func):
```

Comment:
```suggestion
def maybe_wrap_arrays(func):
```

Would be nice to use a more informative name here.